### DISCUSSION TITLE
PHP support?

### DISCUSSION THREAD

**Discussion by yuppox** [cesanta/mongoose#2684] [Ask questions, get answers] [answered]
- My goal is: use mongoose to serve local php files
- My actions were: download mongoose.exe and run
- My expectation was: there would be some way to configure php
- The result I saw: no apparent option for PHP or CGI
- My question is: how can I use mongoose as a simple server with PHP?

# Environment

- mongoose version: 7.12
- Compiler/IDE and SDK: prebuilt
- Target hardware/board: windows
- Connectivity chip/module: N/A
- Target RTOS/OS (if applicable): N/A

**Comment by scaprile** [comment, marked as answer]
Please follow the guidelines in the [documentation](https://mongoose.ws/documentation/),  [examples](https://github.com/cesanta/mongoose/tree/master/examples/) and [tutorials](https://mongoose.ws/documentation/#tutorials-and-examples) available.

#2659 

Mongoose is embedded-oriented.

**Reply by yuppox** [reply]
Thanks for the reply. I guess something has changed in the provided executable. I find online help pages that say to use a config file or command-line argument to specify PHP-handling. That doesn't seem to exist.

The documentation, examples and tutorials don't seem to have any mention of external processing (CGI, FastCGI or PHP specifically).

The embedded systems I have are capable of handling PHP and this seems like it would be a common use-case. I was hoping there might be an example or (based on the aforementioned resources) a pre-built solution.

I don't know if I have time to build something by myself but thanks again for your reply and for moving my topic to the proper area. I'll see what I can do.

**Reply by scaprile** [reply]
For "embedded" we mean those dozens of boards that you can see in our docs and don't run operating systems, and access memory in typically less than MB quantities
We are not responsible for what online pages say, if those aren't ours.
We don't provide executables, Mongoose is source code, there is one example that has traditionally been offered in binary form for some people to peruse it, IIRC that is the http-server example.

**Reply by yuppox** [reply]
I didn't imply that you have responsibility for websites. I was just explaining my position, which others are likely to experience.

Your docs include the Raspberry Pi, which certainly has an OS. While many devices are "bare metal", the docs also list multiple ones with an OS. Furthermore, your website has a section for binaries:
https://mongoose.ws/binary/

So neither of your statements seem to be accurate on those topics (supported devices "don't run operating systems" and you "do not provide executables").

I've tried to explain my use case and ask for help. You don't seem to have any interest in helping and seem determined to dismiss my questions without consideration.

**Reply by scaprile** [reply]
Sorry about your feelings.
This explains the one binary you found
> We don't provide executables, Mongoose is source code, there is one example that has traditionally been offered in binary form for some people to peruse it, IIRC that is the http-server example.

The vast majority of our paid customers, which you may find here: https://cesanta.com/about.html, drive our focus. My exact phrase was:
> For "embedded" we mean those dozens of boards that you can see in our docs and don't run operating systems, and access memory in typically less than MB quantities

and holds true.
These customers do not requested the need to support PHP, hence we won't add that, because we don't think that is needed.
Our support for OSs is mainly to provide an easy development path.
People in the OS and Open Source worlds also use Mongoose for other purposes, too.

I'm closing this as I don't think more clarifications are needed and your opening question has been answered.
